.github/actions/setup-linux/action.yml

```diff
@@ -0,0 +1,78 @@
+name: 'Setup on Linux'
+description: 'Installs dependencies and sets env variables specific to Linux'
+inputs:
+  codec-aom:
+    description: 'Can take the values: OFF, LOCAL, SYSTEM'
+    default: 'OFF'
+  codec-dav1d:
+    description: 'Can take the values: OFF, LOCAL, SYSTEM'
+    edfault: 'OFF'
+  codec-rav1e:
+    description: 'Can take the values: OFF, LOCAL, SYSTEM'
+    default: 'OFF'
+  extra-cache-key:
+    description: 'Extra cache key to use in the cache name. Useful when several caches are used in one workflow.'
+    default: ''
+  gcc-version:
+    description: 'Can be empty, in which case CC and CXX are not overriden'
+    default: ''
+  libxml2:
+    default: 'OFF'
+  libyuv:
+    default: 'OFF'
+  recent-cmake:
+    description: 'Can take the values: true, false'
+    default: 'false'
+outputs:
+  ext-cache-hit:
+    value: ${{ steps.cache.outputs.ext-cache-hit }}
+runs:
+  using: "composite"
+  steps:
+    - name: Install non-library dependencies
+      run: |
+        sudo apt update
+        sudo apt install imagemagick libjpeg-turbo8-dev libpng-dev
+      shell: bash
+    - name: Install libaom library
+      if: ${{ inputs.codec-aom == 'SYSTEM' }}
+      run:
+        sudo apt install libaom-dev
+      shell: bash
+    - name: Install libdav1d library
+      if: ${{ inputs.codec-dav1d == 'SYSTEM' }}
+      run:
+        sudo apt install libdav1d-dev
+      shell: bash
+    - name: Install rav1e library
+      if: ${{ inputs.codec-rav1e == 'SYSTEM' }}
+      run:
+        sudo apt install librav1e-dev
+      shell: bash
+    - name: Install libxml2 library
+      if: ${{ inputs.libxml2 == 'SYSTEM' }}
+      run:
+        sudo apt install libxml2
+      shell: bash
+    - name: Install libyuv library
+      if: ${{ inputs.libyuv == 'SYSTEM' }}
+      run:
+        sudo apt install libyuv-dev
+      shell: bash
+
+    - uses: ./.github/actions/setup-common
+      with:
+        codec-aom: ${{ inputs.codec-aom }}
+        codec-dav1d: ${{ inputs.codec-dav1d }}
+        codec-rav1e: ${{ inputs.codec-rav1e }}
+        recent-cmake: ${{ inputs.recent-cmake }}
+    - uses: ./.github/actions/cache
+      id: cache
+      with:
+        extra-key: ${{ inputs.extra-cache-key }}
+        use-rust: ${{ inputs.codec-rav1e == 'LOCAL' }}
+
+    - name: Set GCC & G++ compiler
+      if: ${{ inputs.gcc-version != '' }}
+      run: echo "CC=gcc-${{ inputs.gcc-version }}" >> $GITHUB_ENV && echo "CXX=g++-${{ inputs.gcc-version }}" >> $GITHUB_ENV
+      shell: bash
```

Comment:
Here it matters which shell we are using, but bash is very likely the default.

Reply:
Any shell matters as long as GITHUB_ENV is filled.

Reply:
My previous comment may not be clear. I meant that in this step the shell has to be bash because the command uses bash's syntax.